plugins/inputs/sql/sql.go

```diff
@@ -375,6 +377,13 @@ func (s *SQL) setupConnection() error {
 	dsn := string(dsnSecret)
 	config.ReleaseSecret(dsnSecret)
 
+	// Replaces special characters in the oracle connection string
+	if s.driverName == "oracle" {
+		escaped := url.PathEscape(dsn)
+		dsn = strings.ReplaceAll(escaped, "%2F", "/")
+		s.Log.Debugf("Connection %#v", dsn)
```

Comment:
Please remove this debugging statement as it has the potential to leak sensitive information encoded in the DSN like credentials...

Reply:
Changes made